### ISSUE TITLE
 away3d-core-fp11 regression i.e. i = ( 1 == 1) ? num : val == val ? val : 0 ;  // Should not be an error CS0172

### ISSUE THREAD

**Issue by sushihangover** [PlayScriptRedux/playscript#56]
This is a regression found in the away3d-core-fp11:

away3d-core-fp11/src/away3d/animators/PathAnimator.as(337,51): error CS0172: Type of conditional expression cannot be determined as `uint' and`int' convert implicitly to each other

```
_index = (val > _path.numSegments - 1) ? _path.numSegments - 1 : (val > 0) ? val : 0;
```

This should not report an CS0172:

i = ( 1 == 1) ? num : val == val ? val : 0 ;

**Comment by sushihangover** [comment]
In C#, this would be an issue, but allowed in AS

[CS0172](https://msdn.microsoft.com/en-us/library/1yaky37f%28v=vs.90%29.aspx)

```
// CS0172.cs
public class Square
{
   public class Circle
   {
      public static implicit operator Circle(Square aa)
      {
         return null;
      }

      public static implicit operator Square(Circle aa)
      // using explicit resolves this error
      // public static explicit operator Square(Circle aa)
      {
         return null;
      }
   }

   public static void Main()
   {
      Circle aa = new Circle();
      Square ii = new Square();
      object o = (1 == 1) ? aa : ii;   // CS0172
      // the following cast would resolve this error
      // (1 == 1) ? aa : (Circle)ii;
   }
}
```

**Comment by sushihangover** [comment]
New AS test to debug regression:

as/test-as-CS0172.as

```
// Compiler options: -psstrict+
package {
    public class Foo {
        public static function Main():int {
                var val:uint = 1;
                var num:uint = 1;
                var i:uint;

                // The folliwing SHOULD NOT report this:

                // error CS0172: Type of conditional expression cannot be determined
                // as `uint' and `int' convert implicitly to each other

                // but should in C#

                i = ( 1 == 1) ? num : val ;  // not a problem
                trace(i); // 1
                i = ( 1 == 1) ? num : val == val ? val : 0 ; // error CS0172
                trace(i); // 1
                i = ( 1 == 1) ? num - 1 : (val > 0) ? val : 0; // error CS0172
                trace(i); // 0
                return 0;
        }
    }
}
```

**Comment by sushihangover** [comment]
commit c026480bb94c397c4d29cf4c2af3c43f6d1e1840
Author: SushiHangover sushihangover@outlook.com
Date:   Fri Aug 14 17:14:29 2015 -0700

```
CS0172 implicit boolean conversion in conditional expressions allowed in AS
Closes #56 :
away3d-core-fp11 regression i.e. i = ( 1 == 1) ? num : val == val ? val : 0 ; i
```
